src/io/flutter/editor/WidgetIndentsPass.java

```diff
@@ -0,0 +1,783 @@
+/*
+ * Copyright 2019 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+package io.flutter.editor;
+
+import com.intellij.openapi.editor.*;
+import com.intellij.openapi.editor.colors.EditorColors;
+import com.intellij.openapi.editor.colors.EditorColorsScheme;
+import com.intellij.openapi.editor.ex.EditorEx;
+import com.intellij.openapi.editor.markup.*;
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Key;
+import com.intellij.openapi.util.Segment;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.ui.Gray;
+import com.intellij.ui.JBColor;
+import com.intellij.ui.paint.LinePainter2D;
+import com.intellij.util.DocumentUtil;
+import com.intellij.util.text.CharArrayUtil;
+import com.jetbrains.lang.dart.analyzer.DartAnalysisServerService;
+import io.flutter.settings.FlutterSettings;
+import org.dartlang.analysis.server.protocol.FlutterOutline;
+import org.jetbrains.annotations.NotNull;
+
+import java.awt.*;
+import java.util.*;
+import java.util.List;
+
+import static java.lang.Math.*;
+
+/**
+ * Data describing widget indents for an editor that is persisted across
+ * multiple runs of the WidgetIndentsPass.
+ */
+class WidgetIndentsPassData {
```

Comment:
I'd put the public class in this file first.

Reply:
done